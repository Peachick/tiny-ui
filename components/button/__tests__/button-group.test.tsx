import React from 'react';
import { render, mount } from 'enzyme';
import Button from '../button';
import ButtonGroup from '../button-group';

describe('<ButtonGroup />', () => {
  it('should match the snapshot', () => {
    const wrapper = render(
      <ButtonGroup>
        <Button>Default</Button>
      </ButtonGroup>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render child button correctly', () => {
    const wrapper = mount(
      <ButtonGroup btnType="primary">
        <Button>Default</Button>
      </ButtonGroup>
    );
    expect(wrapper.childAt(0).find('.ty-btn_primary').length).toBe(1);
  });
});
