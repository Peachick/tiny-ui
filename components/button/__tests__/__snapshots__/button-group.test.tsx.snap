// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonGroup /> should match the snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "ty-btn-group ty-btn-group_default",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "ty-btn ty-btn_default ty-btn_md",
          "role": "button",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "ty-btn__children",
            },
            "children": Array [
              Object {
                "data": "Default",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "button",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "role": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "role": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
